contrib/collection.cgi

```diff
@@ -61,6 +65,27 @@ for (qw(action host plugin plugin_instance type type_instance timespan))
 
 exit (main ());
 
+sub read_types_db
+{
+	my $fh;
+	open ($fh, "< $TypesDB") or confess ("open ($TypesDB): $!");
+	while (my $line = <$fh>)
+	{
+		chomp ($line);
+		$line =~ s/\s*(?:#.*)?$//;
+		next if (!$line);
+		my ($type, $specs) = split(m/\s+/, $line, 2);
+		my @specs = split(m/,?\s+/, $specs);
+		$Types->{$type} = [];
+		for (my $i = 0; $i < @specs; $i++)
```

Comment:
I suggest iterating over the elements itself, like
```perl
foreach my $spec (@specs)
{
  my ($name, $ds_type, $min, $max) = split(m/:/, $specs);
  ...
}

Reply:
I think you meant `split(..., $spec)` rather than `split(..., $specs)`.

I've made the change (but I'm not currently set up to test that it works ;).